pkg/agent/openflow/pipeline.go

```diff
@@ -460,19 +463,31 @@ func (c *client) Delete(flow binding.Flow) error {
 	return c.DeleteAll([]binding.Flow{flow})
 }
 
+func fillActions(l int, act OFAction) []OFAction {
+	acts := make([]OFAction, 0, l)
+	for i := 0; i < l; i++ {
+		acts = append(acts, act)
+	}
+	return acts
+}
+
 func (c *client) AddAll(flows []binding.Flow) error {
-	return c.changeAll(flows, add)
+	return c.changeAll(map[OFAction][]binding.Flow{add: flows})
 }
 
 func (c *client) ModifyAll(flows []binding.Flow) error {
-	return c.changeAll(flows, mod)
+	return c.changeAll(map[OFAction][]binding.Flow{mod: flows})
 }
 
 func (c *client) DeleteAll(flows []binding.Flow) error {
-	return c.changeAll(flows, del)
+	return c.changeAll(map[OFAction][]binding.Flow{del: flows})
+}
+
+func (c *client) HardModifyAll(flowsMap map[OFAction][]binding.Flow) error {
```

Comment:
ditto, I feel HardModifyAll would introduce misunderstanding. Ideally we just need add, modify, and delete.

Reply:
Updated.